path_login.go

```diff
@@ -295,13 +294,13 @@ func (b *azureAuthBackend) verifyResource(ctx context.Context, subscriptionID, r
 			if err != nil {
 				return fmt.Errorf("unable to create msi client: %w", err)
 			}
-			userIdentity, err := msiClient.Get(ctx, msiResourceGroupName, msiResourceName)
+			userIdentityResponse, err := msiClient.Get(ctx, msiResourceGroupName, msiResourceName, nil)
 			if err != nil {
 				return fmt.Errorf("unable to retrieve user assigned identity metadata: %w", err)
 			}
 
-			if userIdentity.PrincipalID != nil {
-				principalIDs[userIdentity.PrincipalID.String()] = struct{}{}
+			if userIdentityResponse.Properties.PrincipalID != nil {
+				principalIDs[*userIdentity.PrincipalID] = struct{}{}
 			}
```

Comment:
Ah, I think the principal ID we're adding here is wrong. It should be the one returned from the `msiClient.Get()` call in the VMSS flex case. Also it's missing a nil check on `userIdentityResponse.Properties` (not likely, but we should be defensive in any case).

Let's change this to:
```suggestion
			if userIdentityResponse.Properties != nil && userIdentityResponse.Properties.PrincipalID != nil {
				principalIDs[*userIdentityResponse.Properties.PrincipalID] = struct{}{}
			}
```

Reply:
Nice, was unsure about this so thanks for catching it!